src/main/java/net/sf/jabref/gui/maintable/SpecialMainTableColumns.java

```diff
@@ -43,45 +39,45 @@ public String getDisplayName() {
         public Object getColumnValue(BibEntry entry) {
 
             return entry.getField(SpecialFields.FIELDNAME_RANKING)
-                    .flatMap(Rank.getInstance()::parse).map(rank -> rank.createLabel()).orElse(null);
+                    .flatMap(SpecialField.RANK::parse).map(rank -> SpecialFieldIcon.createSpecialFieldValueLabel(rank)).orElse(null);
         }
     };
 
     public static final MainTableColumn PRIORITY_COLUMN = new MainTableColumn(SpecialFields.FIELDNAME_PRIORITY,
```

Comment:
Well, the columns differ somewhat (multi-value, versus single-value), so not in an entirely beautiful way. But I turned the whole thing into a package-private non-static builder class, so it is much less ugly.